src/bootstrap/render_tests.rs

```diff
@@ -41,7 +41,7 @@ pub(crate) fn try_run_tests(builder: &Builder<'_>, cmd: &mut Command) -> bool {
     }
 }
 
-fn run_tests(builder: &Builder<'_>, cmd: &mut Command) -> bool {
+fn run_tests(builder: &Builder<'_>, cmd: &mut Command, stream: bool) -> bool {
     cmd.stdout(Stdio::piped());
```

Comment:
Could we, instead of reading-and-writing each byte of stdout, just not call this if `stream` is set, so that stdout is inherited?

Reply:
The whole idea is implementing this functionality to `Renderer` which can be used from the test tools. If we don't run the `run_tests`, then we don't need to use `try_run_tests` either. And that sounds wrong in my opinion

Reply:
In `rustdoc-gui-test`, we are re-writing on the same line for each test. Currently `Renderer` is writing to stdout only when the line is completed, which seems weird.

Reply:
Sorry, might have highlighted the wrong line - I was referring to the `cmd.stdout` call. Unless I'm misunderstanding and the stdout-writing needs to exist on a method on `Renderer` - in that case I think this is fine

Reply:
> might have highlighted the wrong line

Actually you pointed the correct line, but I focused on the highlighted line directly. My bad.


`cmd.stdout(Stdio::piped());` has to stay like that, otherwise the `Renderer` will panic and release the stdin and keep writing to stdout. And the process will be uninterruptible(e.g. with CTRL-C) which means we will have to wait until the child process finishes.